File: java/Conversor.java
import java.io.PrintStream;
import java.nio.charset.StandardCharsets;

public class Conversor {

    static long converterParaDecimal(String valorHex) {
        return Long.parseLong(valorHex, 16);
    }

    public static void main(String[] args) {
        System.setOut(new PrintStream(System.out, true, StandardCharsets.UTF_8));

        if (args.length != 1) {
            System.out.println("Uso: java Conversor <hexadecimal>");
            System.exit(1);
        }

        String entrada = args[0];
        String valorHex = entrada.trim();
        if (valorHex.toLowerCase().startsWith("0x")) {
            valorHex = valorHex.substring(2);
        }

        long numeroDecimal;
        try {
            numeroDecimal = converterParaDecimal(valorHex);
        } catch (NumberFormatException e) {
            System.out.println("Erro: '" + entrada + "' não é um hexadecimal válido.");
            System.exit(1);
            return;
        }

        System.out.println("Hexadecimal: " + entrada.toUpperCase());
        System.out.println("Decimal:     " + numeroDecimal);
        System.out.println("Binário:     " + Long.toBinaryString(numeroDecimal));
        System.out.println("Octal:       " + Long.toOctalString(numeroDecimal));
    }
}

File: javascript/conversor.js
function converterHex(valorHex) {
    valorHex = valorHex.trim().toLowerCase().replace(/^0x/, "");

    const numeroDecimal = parseInt(valorHex, 16);

    if (Number.isNaN(numeroDecimal)) {
        throw new Error(`'${valorHex}' não é um hexadecimal válido.`);
    }

    return {
        decimal: numeroDecimal,
        binario: numeroDecimal.toString(2),
        octal: numeroDecimal.toString(8),
    };
}

function main() {
    const argumento = process.argv[2];

    if (!argumento) {
        console.log("Uso: node conversor.js <hexadecimal>");
        process.exit(1);
    }

    try {
        const resultado = converterHex(argumento);
        console.log(`Hexadecimal: ${argumento.toUpperCase()}`);
        console.log(`Decimal:     ${resultado.decimal}`);
        console.log(`Binário:     ${resultado.binario}`);
        console.log(`Octal:       ${resultado.octal}`);
    } catch (erro) {
        console.log(`Erro: ${erro.message}`);
        process.exit(1);
    }
}

main();
